public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -162,6 +162,44 @@ export class PanelModel implements DataConfigSource {
       (this as any)[property] = model[property];
     }
 
+    const notPersistedPropertiesKeys = Object.keys(notPersistedProperties);
+    const mustKeepPropsKeys = Object.keys(mustKeepProps);
+    const defaultsKeys = Object.keys(defaults);
+    const properties = Object.keys(this);
+    for (const property of properties) {
+      if (notPersistedPropertiesKeys.find(prop => property === prop)) {
```

Comment:
any reason for not using index lookup?  notPersistedProperties, mustKeepProps & defaults are objects for a reason to be able to do index-based lookup 

if (notPersistedProperties[property])

Reply:
Of course, great catch 🙈

Reply:
Done